### ISSUE TITLE
recipe for CommonVoice

### ISSUE THREAD

**Issue by argideritzalpea** [NVIDIA/NeMo#568]
I noticed that one of the main models was trained with both Librispeech and CommonVoice. Is there a recipe / set of tools for downloading the common voice data?

**Comment by okuchaiev** [comment]
You can download Mozilla Common Voice dataset here https://voice.mozilla.org/en/datasets 
Note that it has several languages, so chose the one you need.

To use it with NeMo, you'll need to convert mp3 files into wav and create NeMo ASR data layer manifests from ".tsv" files provided by Mozilla. Here is an example script on how to do this ('convert_tsvs_to_manifest.py'):

```python
import sys
import os
import pandas as pd
import subprocess
import json
import codecs
import unidecode


def normalize_str(txt) -> str:
    # TODO: REPLACE WITH YOUR OWN NORMALIZATION LOGIC HERE!!!!   
    valid_chars = (" ", "'", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z")
    new_txt = unidecode.unidecode(txt.lower().strip())
    res_arr = []
    for c in new_txt:
        if c in valid_chars:
            res_arr.append(c)
        else:
            res_arr.append(' ')
    res = ''.join(res_arr).strip()    
    return ' '.join(res.split())

def tsv_to_manifest(tsv_files, manifest_file, prefix):
  manifests = []
  for tsv_file in tsv_files:
    print('Processing: {0}'.format(tsv_file))
    dt = pd.read_csv(tsv_file, sep='\t', encoding='utf8')
    for index, row in dt.iterrows():
      try:
        entry = {}
        os.system("mkdir -p wavs/{0}".format(prefix))
        mp3_file = "clips/" + row['path'] # + ".mp3"
        wav_file = "wavs/{0}/".format(prefix) + row['path'] + ".wav"
        subprocess.check_output("sox {0} -c 1 -r 16000 {1}".format(mp3_file, wav_file), shell=True)
        duration = subprocess.check_output(
          "soxi -D {0}".format(wav_file), shell=True)
        entry['audio_filepath'] = wav_file
        entry['duration'] = float(duration)
        entry['text'] = normalize_str(row['sentence'])
        manifests.append(entry)
      except:
        print("SOMETHING WENT WRONG - IGNORING ENTRY")

  with codecs.open(manifest_file, 'w', encoding='utf-8') as fout:
    for m in manifests:
      fout.write(json.dumps(m, ensure_ascii=False) + '\n')
  print('Done!')


def main():
  prefix = sys.argv[1]
  tsv_to_manifest([prefix + ".tsv"], prefix+".json", prefix)
  

if __name__ == "__main__":
    main()
```
On a linux system (works on Ubuntu 18 with soxi installed) you can you it like this:
``python convert_tsvs_to_manifest.py train`` to convert ' train.tsv' to 'train.json' which can be consumed by NeMo

**Comment by ycstone** [comment]
Do you mind to share what are the values for learning rate, epochs and batch? Thanks a lot.
